tests/testthat/test-add_file.R

```diff
@@ -0,0 +1,232 @@
+test_that("add_file finds test files and adds an evaluation section to the current report", {
+  withr::with_tempdir({
+
+    vt_use_validation_config()
+    vt_use_validation()
+    vt_use_test_code("example_code.r",username = "sample")
+
+    file.create("report.Rmd")
+    file.create("report_dynamic.Rmd")
+    file.create("report2.Rmd")
+    file.create("report_dynamic2.Rmd")
+
+    test_code <- validation_file_path(
+      file = "example_code.r",
+      ref = vt_path()
+    )
+
+    add_file.r_test_code(test_code,report = "report.Rmd")
+    add_file.r_test_code(test_code,report = "report_dynamic.Rmd", dynamic_referencing = TRUE)
+    add_file(test_code,  report = "report2.Rmd")
+    add_file(test_code,report = "report_dynamic2.Rmd", dynamic_referencing = TRUE)
+
+    report_text <- readLines("report.Rmd")
+    report_dynamic_text <- readLines("report_dynamic.Rmd")
+    report_text2 <- readLines("report2.Rmd")
+    report_dynamic_text2 <- readLines("report_dynamic2.Rmd")
+
+    expect_equal(
+      report_text,
+      c(
+        "",
+        "```{r example-code, echo = FALSE}",
+        "results <- vt_run_test_code_file(file=\"example_code.r\")",
+        "vt_kable_test_code(results)",
+        "```"
+      )
+    )
+
+    expect_equal(
+      report_dynamic_text,
+      c(
+        "",
+        "```{r example-code, echo = FALSE}",
+        "results <- vt_run_test_code_file(file=\"example_code.r\")",
+        "results <- dynamic_reference_rendering(results)",
+        "vt_kable_test_code(results)",
+        "```"
+      )
+    )
+
+    expect_equal(
+      report_text2,
+      c(
+        "",
+        "```{r example-code, echo = FALSE}",
+        "results <- vt_run_test_code_file(file=\"example_code.r\")",
+        "vt_kable_test_code(results)",
+        "```"
+      )
+    )
+
+    expect_equal(
+      report_dynamic_text2,
+      c(
+        "",
+        "```{r example-code, echo = FALSE}",
+        "results <- vt_run_test_code_file(file=\"example_code.r\")",
+        "results <- dynamic_reference_rendering(results)",
+        "vt_kable_test_code(results)",
+        "```"
+      )
+    )
+  })
+})
+
+test_that("add_file_to_report is a simple wrapper around add_file", {
+  withr::with_tempdir({
+
+    vt_use_validation_config()
+    vt_use_validation()
+    vt_use_req("example_req.md",username = "sample")
+    vt_use_test_code("example_code.r",username = "sample")
+
+    file.create("report.Rmd")
+    file.create("report_dynamic.Rmd")
+
+    #standard
+    add_file_to_report(
+      file = "example_req.md",
+      report = "report.Rmd")
+    add_file_to_report(
+      file = "example_code.r",
+      report = "report.Rmd")
+
+    #dynamic
+    add_file_to_report(
+      file = "example_req.md",
+      report = "report_dynamic.Rmd",
+      dynamic_referencing = TRUE)
+    add_file_to_report(
+      file = "example_code.r",
+      report = "report_dynamic.Rmd",
+      dynamic_referencing = TRUE)
+
+
+    report_text <- readLines("report.Rmd")
```

Comment:
knitting this `report.Rmd` document does not preserve markdown formatting

![image](https://user-images.githubusercontent.com/253189/112375473-935e0300-8ca0-11eb-9ea9-ffa1fe05d2c7.png)

Reply:
🤔 interesting. It should be returning "asis". I'll look into it